File: apps/mobile_flutter/ios/Runner/Video/YohPalVideoPlugin.swift
import Flutter
import UIKit

class YohPalVideoPlugin: NSObject, FlutterPlugin {
    private let videoView = YohPalVideoView()

    static func register(with registrar: FlutterPluginRegistrar) {
        let instance = YohPalVideoPlugin()
        registrar.register(
            YohPalVideoViewFactory(videoView: instance.videoView),
            withId: "yohpal.video/view"
        )
        let channel = FlutterMethodChannel(
            name: "yohpal.video/native",
            binaryMessenger: registrar.messenger()
        )
        registrar.addMethodCallDelegate(instance, channel: channel)
    }

    func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        let args = call.arguments as? [String: Any]
        switch call.method {
        case "load":
            if let url = args?["url"] as? String { videoView.load(url: url) }
            result(nil)
        case "play":
            videoView.play()
            result(nil)
        case "pause":
            videoView.pause()
            result(nil)
        case "seekTo":
            videoView.seekTo(ms: args?["ms"] as? Int ?? 0)
            result(nil)
        case "setMuted":
            videoView.setMuted(args?["muted"] as? Bool ?? false)
            result(nil)
        case "dispose":
            videoView.dispose()
            result(nil)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}

File: apps/mobile_flutter/android/app/src/main/kotlin/com/yohpal/live/video/YohPalVideoPlugin.kt
package com.yohpal.live.video

import androidx.media3.common.util.UnstableApi
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodChannel

@UnstableApi
class YohPalVideoPlugin : FlutterPlugin {
    private lateinit var channel: MethodChannel
    private var videoView: YohPalVideoView? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        videoView = YohPalVideoView(binding.applicationContext)
        binding.platformViewRegistry.registerViewFactory(
            "yohpal.video/view",
            YohPalVideoViewFactory { videoView!! }
        )
        channel = MethodChannel(binding.binaryMessenger, "yohpal.video/native")
        channel.setMethodCallHandler { call, result ->
            when (call.method) {
                "load" -> {
                    val url = call.argument<String>("url")!!
                    videoView?.load(url)
                    result.success(null)
                }
                "play" -> {
                    videoView?.play()
                    result.success(null)
                }
                "pause" -> {
                    videoView?.pause()
                    result.success(null)
                }
                "seekTo" -> {
                    val ms = call.argument<Int>("ms") ?: 0
                    videoView?.seekTo(ms.toLong())
                    result.success(null)
                }
                "setMuted" -> {
                    val muted = call.argument<Boolean>("muted") ?: false
                    videoView?.setMuted(muted)
                    result.success(null)
                }
                "dispose" -> {
                    videoView?.dispose()
                    result.success(null)
                }
                else -> result.notImplemented()
            }
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
        videoView?.dispose()
        videoView = null
    }
}
